src/instana/instrumentation/asgi.py

```diff
@@ -4,106 +4,135 @@
+            request_context = tracer.extract(Format.BINARY, request_headers)
 
-        async def send_wrapper(response):
-            span = async_tracer.active_span
-            if span is None:
-                await send(response)
-            else:
-                if response['type'] == 'http.response.start':
-                    try:
-                        status_code = response.get('status')
-                        if status_code is not None:
-                            if 500 <= int(status_code):
-                                span.mark_as_errored()
-                            span.set_tag('http.status_code', status_code)
-
-                        headers = response.get('headers')
-                        if headers is not None:
-                            self._extract_custom_headers(span, headers)
-                            async_tracer.inject(span.context, opentracing.Format.BINARY, headers)
-                    except Exception:
-                        logger.debug("send_wrapper: ", exc_info=True)
+        with tracer.start_as_current_span("asgi", span_context=request_context) as span:
+            self._collect_kvs(scope, span)
+            if "headers" in scope and agent.options.extra_http_headers:
+                self._extract_custom_headers(span, scope["headers"])
 
-                try:
-                    await send(response)
-                except Exception as exc:
-                    span.log_exception(exc)
-                    raise
-
-        with async_tracer.start_active_span("asgi", child_of=request_context) as tracing_scope:
-            self._collect_kvs(scope, tracing_scope.span)
-            if 'headers' in scope and agent.options.extra_http_headers is not None:
-                self._extract_custom_headers(tracing_scope.span, scope['headers'])
+            instana_send = self._send_with_instana(
+                span,
+                scope,
+                send,
+            )
 
             try:
-                await self.app(scope, receive, send_wrapper)
+                await self.app(scope, receive, instana_send)
             except Exception as exc:
-                tracing_scope.span.log_exception(exc)
+                span.record_exception(exc)
                 raise exc
+
+    def _send_with_instana(
+        self,
+        current_span: "InstanaSpan",
+        scope: Dict[str, Any],
+        send: Callable[[Dict[str, Any]], Awaitable[None]],
+    ) -> Awaitable[None]:
+        async def send_wrapper(response: Dict[str, Any]) -> Awaitable[None]:
+            if response["type"] == "http.response.start":
+                try:
+                    status_code = response.get("status")
+                    if status_code:
+                        if 500 <= int(status_code):
+                            current_span.mark_as_errored()
+                        current_span.set_attribute("http.status_code", status_code)
```

Comment:
Can we use the pre-defined attribute here?

```suggestion
                        current_span.set_attribute(SpanAttributes.HTTP_STATUS_CODE, status_code)
```

Reply:
Done.